src/status_im2/common/device_permissions.cljs

```diff
@@ -0,0 +1,19 @@
+(ns status-im2.common.device-permissions
+  (:require
+    [quo2.foundations.colors :as colors]
+    [utils.i18n :as i18n]
+    [utils.re-frame :as rf]))
+
+(defn camera
+  [callback]
+  (rf/dispatch
+   [:request-permissions
+    {:permissions [:camera]
+     :on-allowed  callback
+     :on-denied   #(rf/dispatch
```

Comment:
In some cases, we need to show a system alert to inform (to enable permission from Settings) and a button to take the user to Settings. It's better if the function receives another callback for a denied case or if we show the alert popup here.

Reply:
We probably need a design update for that. I think we would need to display that alert optionally for all permissions.